### ISSUE TITLE
Error: Inconsistent conditional result types

### ISSUE THREAD

**Issue by pysiok1983** [hashicorp/terraform-provider-google#11283]
Hi,
I am using TF 1.1.0 and 4.8.0 for providers. I have tried to use this script directly from Visual Studio but I am receiving the following error:

```
Error: Inconsistent conditional result types
│
│   on .terraform\modules\apigee-x-core.kms-inst-disk\modules\kms\main.tf line 53, in locals:
│   52:     var.keyring_create
│   53:     ? google_kms_key_ring.default.0
│   54:     : data.google_kms_key_ring.default.0
│     ├────────────────
│     │ data.google_kms_key_ring.default[0] is a object, known only after apply
│     │ google_kms_key_ring.default[0] is a object, known only after apply
│     │ var.keyring_create is a bool, known only after apply
│
│ The true and false result expressions must have consistent types. The given expressions are object and object, respectively.
╵
╷
│ Error: Inconsistent conditional result types
│
│   on .terraform\modules\apigee-x-core.kms-org-db\modules\kms\main.tf line 53, in locals:
│   52:     var.keyring_create
│   53:     ? google_kms_key_ring.default.0
│   54:     : data.google_kms_key_ring.default.0
│     ├────────────────
│     │ data.google_kms_key_ring.default[0] is a object, known only after apply
│     │ google_kms_key_ring.default[0] is a object, known only after apply
│     │ var.keyring_create is a bool, known only after apply
│
│ The true and false result expressions must have consistent types. The given expressions are object and object, respectively.

│ Error: Inconsistent conditional result types
│
│   on .terraform\modules\bridge-mig\modules\compute-mig\main.tf line 149, in locals:
│  148:     var.regional ?
│  149:     google_compute_region_instance_group_manager.default :
│  150:     google_compute_instance_group_manager.default
│     ├────────────────
│     │ google_compute_instance_group_manager.default is a list of object, known only after apply
│     │ google_compute_region_instance_group_manager.default is a list of object, known only after apply
│     │ var.regional is a bool, known only after apply
│
│ The true and false result expressions must have consistent types. The given expressions are list of object and list of object, respectively.
```

**Comment by ggtisc** [comment]
Hi @pysiok1983 

If you are still having the same issue please open up a new ticket sharing your resource(s) code without using locals variables or modules to check if there is something wrong with the Google provider resources. Otherwise if it is an issue related to modules then it depends on the users configuration. I suggest you check locals, variables values, how are you sending the info through modules, etc. For sensitive data you could use examples like:

project = "my-project"
user = "my-user@my-domain.com"
org_id = 1234567890
